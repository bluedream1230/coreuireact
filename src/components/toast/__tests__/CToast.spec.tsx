import * as React from 'react'
import { render /* ,screen */, fireEvent /* ,waitFor */ } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { CToast } from '../../../index'

test('loads and displays CToast component', async () => {
  const { container } = render(<CToast>Test</CToast>)
  expect(container).toMatchSnapshot()
})

test('CToast customize', async () => {
  const { container } = render(
    <CToast
      className="bazinga"
      autohide={false}
      color="warning"
      delay={100}
      key={1}
      visible={true}
      //onDismiss
    >
      Test
    </CToast>,
  )
  expect(container).toMatchSnapshot()
  expect(container.firstChild).toHaveClass('bazinga')
  expect(container.firstChild).toHaveClass('toast')
  expect(container.firstChild).toHaveClass('fade')
  expect(container.firstChild).toHaveClass('bg-warning')
  expect(container.firstChild).toHaveClass('show')
})

test('CToast click on dismiss button', async () => {
  jest.useFakeTimers()
  const onDismiss = jest.fn()
  const { container } = render(
    <CToast
      className="bazinga"
      autohide={false}
      color="warning"
      delay={100}
      key={1}
      visible={true}
      onDismiss={onDismiss}
    >
      Test
    </CToast>,
  )
  expect(container.firstChild).toHaveClass('show')
  expect(onDismiss).toHaveBeenCalledTimes(0)
  const btn = document.querySelector('.btn-close')
  if (btn !== null) {
    fireEvent.click(btn)
  }
  jest.runAllTimers()
  expect(onDismiss).toHaveBeenCalledTimes(1)
  expect(container.firstChild).toBeNull()
  jest.useRealTimers()
})

test('CToast test autohide', async () => {
  jest.useFakeTimers()
  const { container } = render(
    <CToast autohide={true} delay={100} visible={true}>
      Test
    </CToast>,
  )
  expect(container.firstChild).toHaveClass('show')
  jest.runAllTimers()
  expect(container.firstChild).toBeNull()
  jest.useRealTimers()
})
