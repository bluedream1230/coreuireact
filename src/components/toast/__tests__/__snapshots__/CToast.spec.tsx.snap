// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CToast customize 1`] = `
<div>
  <div
    aria-atomic="true"
    aria-live="assertive"
    class="toast fade show bg-warning border-0 bazinga"
    role="alert"
  >
    Test
  </div>
</div>
`;

exports[`loads and displays CToast component 1`] = `
<div>
  <div
    aria-atomic="true"
    aria-live="assertive"
    class="toast fade show"
    role="alert"
  >
    Test
  </div>
</div>
`;
